forest/navigate.py

```diff
@@ -14,37 +14,54 @@
         rdt)
 
 
-class Config(object):
-    """File system navigation using config file
+class Navigator:
+    def __init__(self, config, database):
+        # TODO: Once the idea of a "Group" exists we can avoid using the
+        # config and defer the sub-navigator creation to each of the
+        # groups. This will remove the need for the `_from_group` helper
+        # and the logic in FileSystemNavigator.from_file_type().
+        # Also, it'd be good to switch the identification of groups from
+        # using the `pattern` to using the `label`. In general, not every
+        # group would have a `pattern`.
+        # e.g.
+        # self._navigators = {group.label: group.navigator for group in ...}
+        self._navigators = {group.pattern: self._from_group(group, database)
+                           for group in config.file_groups}
 
-    This implementation performs a glob and then delegates to
-    FileSystem navigators
-    """
-    def __init__(self, config):
-        self.config = config
-        self.navigators = {}
-        for group in self.config.file_groups:
-            if group.directory is None:
-                pattern = group.pattern
-            else:
-                pattern = os.path.join(group.directory, group.pattern)
-            paths = glob.glob(os.path.expanduser(pattern))
-            self.navigators[group.pattern] = FileSystem.file_type(paths, group.file_type)
+    @classmethod
+    def _from_group(cls, group, database):
+        if group.locator == 'database':
+            navigator = database
+        else:
+            paths = cls._expand_paths(group.directory, group.pattern)
+            navigator = FileSystemNavigator.from_file_type(paths,
+                                                           group.file_type)
+        return navigator
+
+    @classmethod
+    def _expand_paths(cls, directory, pattern):
+        if directory is not None:
+            pattern = os.path.join(directory, pattern)
+        return glob.glob(os.path.expanduser(pattern))
```

Comment:
I've not put in a pedantic comment yet. Should this be a `@staticmethod` it doesn't use `cls`?

Reply:
Fair point. I kept it as `@classmethod` just so that the two helper methods used the same decorator, but I'm increasingly inclined to either get rid of both decorators or just make both of them simple functions.

Reply:
I think eventually it will be migrated into a sub-navigator so perhaps not worth contemplating it too much